homeassistant/core.py

```diff
@@ -809,21 +810,38 @@ def _gen_unique_id():
 
     @property
     def services(self):
-        """Return dictionary with per domain a list of available services."""
+        """Return dictionary with per domain a dict of available services."""
         return run_callback_threadsafe(
             self._hass.loop, self.async_services,
         ).result()
 
     @callback
     def async_services(self):
-        """Return dictionary with per domain a list of available services.
+        """Return dictionary with per domain a dict of available services.
 
         This method must be run in the event loop.
         """
         return {domain: {key: value.as_dict() for key, value
                          in self._services[domain].items()}
                 for domain in self._services}
 
+    @property
+    def full_services(self):
+        """Return dict with per domain a dict with the service instances."""
+        return run_callback_threadsafe(
+            self._hass.loop, self.async_full_services,
+        ).result()
+
+    @callback
+    def async_full_services(self):
+        """Return dict with per domain a dict with the service instances.
+
+        This method must be run in the event loop.
+        """
+        return {domain: {key: value for key, value
```

Comment:
Don't do this, this is a waste of resources. Better to just wrap it in `MappingProxyType` so it can't be modified.